lib/pure/httpclient.nim

```diff
@@ -98,6 +98,13 @@
 ## You will also have to compile with ``ssl`` defined like so:
 ## ``nim c -d:ssl ...``.
 ##
+## By default, a set of directories and files from the `ssl_certs <ssl_certs.html>`_
+## module are scanned to validate servers.
+## This can be overridden using the SSL_CERT_FILE and SSL_CERT_DIR environment variables.
+##
+## Setting NIM_SSL_CERT_VALIDATION=insecure will disable certificate verification
```

Comment:
Is this similar to how other applications handle this? Does it really make sense for all Nim applications to use the same env variable?

Reply:
Unfortunately there isn't a standard. Some languages and libraries use an env var, some don't.
The main reason is to provide an obvious way to run tests using SSL certs (hence the warning) without encouraging developers to hardcode CVerifyNone in the codebase.
 
The variable can be ignored by setting CVerifyNone or CVerifyPeer.

Reply:
Seems like a pretty cool idea to make test scripts easier without risking leaving it disabled in production code.

Reply:
@jamiesonbecker I'm not sure if I understand your example. An attacker that can run a SUID binary and control env variables can set HTTP[S]_PROXY and send traffic through a malicious host (or do many other things), but this should not result in gaining root privileges unless it's coupled with a vulnerability in the existent code.
(again, CVerifyPeer would disable the env var)

Reply:
@FedericoCeratto sorry, let me be more clear. A non-root attacker that can run a SUID binary (ie pretty much any user) and control env variables can definitely set an HTTPS proxy and send traffic through a malicious host, BUT that would not risk much,  because it doesn't compromise the cert storage on the local host... the HTTPS proxy is simply relaying encrypted traffic.

However, let's say that someone has written the apt or yum binary in nim and it's SUID root, and you only have user access on the system. Now you can force that binary to connect to a malicious endpoint or MITM and install compromised packages by disabling host checks using the env variable NIM_SSL_CERT_VALIDATION. (that's a somewhat contrived example, but certainly within the realm of possibility if nim becomes more popular as a systems language as Go has.)

Reply:
EDIT: if CVerifyPeer disables the env variable, then this point is moot. :)

Reply:
See update below.